index.bs

```diff
@@ -568,8 +565,7 @@ run these steps:
 	2. Let |queue| be a copy of |observer|'s internal {{[[QueuedEntries]]}}
 		slot.
 	3. Clear |observer|'s internal {{[[QueuedEntries]]}} slot.
-	4. Invoke |callback| with |queue| as the first argument and |observer|
-		as the second argument and <a>callback this value</a>.
+	4. Invoke |callback| with |queue| as the first argument and |observer| as the second argument.
```

Comment:
I don't think this is right? The text was meant to say that the observer should be the callback `this` value as https://html.spec.whatwg.org/#custom-element-reactions:dfn-callback-this-value. This is making stuff less defined. I think it could probably be reworded like:

```
Invoke |callback| with |queue| and |observer| as arguments, and with |observer| as the <a>callback this value</a>.
```

Or something like that?

Reply:
I think this bit of the spec pre-dates my involvement, and I never looked at it closely before. Binding the observer as the 'this' argument doesn't make much sense, since the observer object is already passed as an argument. Chromium's implementation doesn't do this; does Firefox?

Reply:
What is the "this" value in Chromium? This seems to log the observer in both Firefox and Chromium:

```
new IntersectionObserver(function() { console.log("this", this) }).observe(document.documentElement)
```

Reply:
Oh, this is interesting -- the following behavior holds for both chromium and firefox:

> new IntersectionObserver(function() { console.log("this", this) }).observe(document.documentElement)
this
IntersectionObserver {root: null, rootMargin: "0px 0px 0px 0px", thresholds: Array(1), delay: 0, trackVisibility: false}

> new IntersectionObserver(() => { console.log("this", this) }).observe(document.documentElement)
this
Window {parent: Window, opener: null, top: Window, length: 4, frames: Window, …}

The difference, in case you can't spot it, is "function()" as opposed to "() =>". I'm not an ECMAScript expert by any stretch of the imagination, so I have no idea why it works that way. To add to the intrigue:

> new IntersectionObserver((function() { console.log("this.a", this.a) }).bind({a : "hello"})).observe(document.documentElement)
this.a
hello

> new IntersectionObserver((() => { console.log("this.a", this.a) }).bind({a : "hello"})).observe(document.documentElement)
this.a
undefined

I don't understand this at all, but it is at least reassuring that this mysterious behavior appears to be interoperable :) In the chromium code, we don't alter the 'this' binding of the callback at all, and I assume that must be true for firefox as well. I think that's the desired behavior; there's no reason to alter the 'this' binding.

Reply:
OK, I got our resident ECMAScript expert to explain this to me. It appears that both chromium and firefox *are* binding the IntersectionObserver instance to the callback 'this' argument. In the interest of backwards compatibility, we probably shouldn't change this.

I don't then see any problem with the existing spec language, so I'm inclined to just abandon this change, unless you think the existing language could be improved somehow.

Reply:
Yeah, arrow functions capture the `this` value from the outer scope, that's the difference and why you need to use `function() {}` to see this.

Yeah, I'm ok with either closing this, or maybe adding a few commas to clarify that the `this` value is something different, no big deal either way.

Reply:
Done.